api/pkg/s3/datastore/azure/azure.go

```diff
@@ -273,3 +273,12 @@ func (ad *AzureAdapter) AbortMultipartUpload(multipartUpload *pb.MultipartUpload
 func (ad *AzureAdapter) ListParts(listParts *pb.ListParts, context context.Context) (*model.ListPartsOutput, S3Error) {
 	return nil, NoError
 }
+
+func (ad *AzureAdapter) ListBackendObjects(context context.Context, prefix string, limit int64, marker string) (*pb.ListObjectResponse, S3Error) {
```

Comment:
this feature looks like implemented only for AWS...do we really need it if it is not multi cloud enabled ? that itself brings in multiple design considerations and complications (to load objects from multiple provider buckets into a single store).
If it is only AWS, i'd rather run my analytics tool directly on the cloud, using my s3 bucket....what is the advantage added by OpenSDS ? Thoughts ?

Reply:
This is the first step, only AWS EMR supported, we can add other cloud in the future.